controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -2860,27 +3311,47 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
             federatedGraphTargetId: graph.targetId,
           });
 
-          const blobStorageDirectory = `${authContext.organizationId}/${graph.id}`;
-          await opts.blobStorage.removeDirectory({ key: blobStorageDirectory });
+          const deletedGraphs: FederatedGraphDTO[] = [];
+
+          // Delete downstream contracts
+          const contracts = await contractRepo.bySourceFederatedGraphId(graph.id);
+          for (const contract of contracts) {
+            const contractGraph = await fedGraphRepo.byId(contract.downstreamFederatedGraphId);
+            if (!contractGraph) {
+              continue;
+            }
+
+            await fedGraphRepo.delete(contractGraph.targetId);
+            deletedGraphs.push(contractGraph);
+          }
 
           await fedGraphRepo.delete(graph.targetId);
 
+          deletedGraphs.unshift(graph);
+
+          for (const deletedGraph of deletedGraphs) {
+            const blobStorageDirectory = `${authContext.organizationId}/${deletedGraph.id}`;
+            await opts.blobStorage.removeDirectory({ key: blobStorageDirectory });
```

Comment:
Put a try/catch around and log the error so we get a notification when data could not be deleted.